### DISCUSSION TITLE
PIO fatal error: Attached buffer is too small

### DISCUSSION THREAD

**Discussion by lxu16** [E3SM-Project/E3SM#6068] [E3SM model help]
I tried to use the 22 nodes to run the E3SMv2 including the ChemUCI chemistry module.  It seems fine for the 5-days test run. When I increase the simulation time to 1-month. The following errors come out when the program tries to output the restart file for the EAM component.

![Screen Shot 2023-11-14 at 9 25 59 AM](https://github.com/E3SM-Project/E3SM/assets/32209342/3e801583-2ee5-485c-bc06-c6f8c0c0ca75)

**Comment by rljacob** [comment]
@dqwu may be able to help.

**Reply by lxu16** [reply]
BTW, I used the compset "F20TR_chemUCI-Linozv3" and resolution is "ne30pg2_r05_oECv3". I also switch on the namelist for the UCI gas chemistry output.

 history_chemdyg_summary = .true.
 history_UCIgaschmbudget_2D = .true.
 history_UCIgaschmbudget_2D_levels = .true.
 history_gaschmbudget_2D = .true.
 history_gaschmbudget_2D_levels = .true.

**Comment by dqwu** [comment]
@lxu16 You can try increasing the default buffer size limit (64 MB) to 128 MB:
xmlchange PIO_BUFFER_SIZE_LIMIT=134217728

**Comment by lxu16** [comment]
@dqwu Thanks for the suggestions! Now it works!
